[CNI] Document 1337 UID workaround for init container. (#10655)
istio/istio.io: content/en/docs/setup/additional-setup/cni/index.md

1. The Istio sidecar proxy starts in the pod along with the pod's other containers.

Init containers execute before the sidecar proxy starts, which can result in traffic loss during their execution.
Avoid this traffic loss with one of the following settings:

1. Set the `uid` of the init container to `1337` using `runAsUser`.
  `1337` is the [`uid` used by the sidecar proxy](/docs/ops/deployment/requirements/#pod-requirements).

>    Traffic sent by this `uid` will not be captured by the Istio's `iptables` rule.
>    Application container traffic will still be captured as usual.

Reviewer: ```suggestion
   Traffic sent by this `uid` is not captured by the Istio's `iptables` rule.
   Application container traffic will still be captured as usual.
```

Init containers execute before the sidecar proxy starts, which can result in traffic loss during their execution.

> Avoid this traffic loss with one or both of the following settings:

Reviewer 2: ```suggestion
Avoid this traffic loss with one of the following settings:
```

there are now three

> * Set the user of init container to `1337` using `runAsUser`. This will make traffic from init container bypass Istio iptables capture.
>   Application container traffic will still be captured as usual.
> * Set the `traffic.sidecar.istio.io/excludeOutboundIPRanges` annotation to disable redirecting traffic to any
>   CIDRs the init containers communicate with.
> * Set the `traffic.sidecar.istio.io/excludeOutboundPorts` annotation to disable redirecting traffic to the

Reviewer 2: ```suggestion
# Set the uid of the init container to `1337` using `runAsUser`. This will make traffic from the init container bypass Istio's `iptables` capture.
  Application container traffic will still be captured as usual.
# Set the `traffic.sidecar.istio.io/excludeOutboundIPRanges` annotation to disable redirecting traffic to any
  CIDRs the init containers communicate with.
# Set the `traffic.sidecar.istio.io/excludeOutboundPorts` annotation to disable redirecting traffic to the
```

changing bullets to numbers as you're saying "only #1 will work" below

  specific outbound ports the init containers use.

> If [DNS proxying](/docs/ops/configuration/traffic-management/dns-proxy/) is enabled,
> and init container sends traffic with host name and needs DNS resolution, only the first workaround listed above will work.

Reviewer 2: ```suggestion
{{< tip >}}
You must use the `runAsUser 1337` workaround if [DNS proxying](/docs/ops/configuration/traffic-management/dns-proxy/) is enabled, and an init container sends traffic to a host name which requires DNS resolution.
{{< /tip >}}
```

{{< warning >}}

> Please use the above traffic capture exclusion annotations with caution, since the IP/port exclusion annotations not only apply to init container traffic,

Reviewer 2: ```suggestion
Please use traffic capture exclusions with caution, since the IP/port exclusion annotations not only apply to init container traffic,
```

> Please use the above traffic capture exclusion annotatioins with caution, since the IP/port exclusion annotations not only apply to init container traffic,

Reviewer 3: ```suggestion
Please use the above traffic capture exclusion annotations with caution, since the IP/port exclusion annotations not only apply to init container traffic,
```
